### PR TITLE
Modify `get_first_partition_window` to account for offset

### PR DESCRIPTION
**Pull request by clairelin135**
Causes a graphQL error reported by a user here: https://dagster.slack.com/archives/C01U954MEER/p1677064207587029

Previously, `get_first_partition_window` did not account for the end offset.

This led to errors such as `get_first_partition_window` returning `None` when the start time window is the current time window and offset > 0.

### CODE REVIEW

File: python_modules/dagster/dagster/_core/definitions/time_window_partitions.py [hunk 1]
```diff
@@ -370,17 +370,41 @@ def get_prev_partition_window(self, start_dt: datetime) -> Optional[TimeWindow]:
...
 
         time_window = next(iter(self._iterate_time_windows(self.start)))
-        if time_window.end.timestamp() <= current_timestamp:
-            return time_window
+
+        if self.end_offset == 0:
+            return time_window if time_window.end.timestamp() <= current_timestamp else None
         else:
-            return None
+            if self.end_offset > 0:
```
Comment (sryza) [inline python_modules/dagster/dagster/_core/definitions/time_window_partitions.py:386]: Nitpick: making this an `elif` would save a level of indentation and make this a little easier to read

File: python_modules/dagster/dagster_tests/definitions_tests/test_time_window_partitions.py [hunk 2]
```diff
@@ -652,3 +652,77 @@ def my_partitioned_config_2(_start, _end):
...
+
+    assert DailyPartitionsDefinition(
+        start_date="2023-01-01", end_offset=1
+    ).get_first_partition_window(
+        current_time=datetime.strptime("2023-01-01", "%Y-%m-%d")
+    ) == time_window(
+        "2023-01-01", "2023-01-02"
+    )
+
+    assert (
```
Comment (sryza) [inline python_modules/dagster/dagster_tests/definitions_tests/test_time_window_partitions.py:670]: Curious why this isn't line-wrapping

**Review by sryza — COMMENTED** [review]
I think it would be worthwhile to include a test where the current_time doesn't lie exactly on a partition boundary.

**Comment by clairelin135** [comment]
@sryza There are a couple of tests in there with `MonthlyPartitionsDefinitions` where `current_time` is in the middle of the month

**Review by sryza — APPROVED** [review]
To be extra careful, I think it's worth having one in the middle of the day or hour. I won't block this though.

Thanks for fixing btw.
